bpf-map-pressure-exporter/src/fetcher.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"errors"
+	"os"
+	"strings"
+
+	"github.com/cilium/ebpf"
+	"github.com/cybozu-go/log"
+)
+
+type IBpfMapPressureFetcher interface {
+	Fetch() []bpfMapPressure
+}
+
+type bpfMapPressureFetcher struct {
+	mapNameStrings []string
+}
+
+type bpfMapPressure struct {
+	mapId       uint32
+	mapName     string
+	mapPressure float64
+}
+
+func newFetcher(mapNameStrings []string) *bpfMapPressureFetcher {
+	return &bpfMapPressureFetcher{
+		mapNameStrings: mapNameStrings,
+	}
+}
+
+func (f *bpfMapPressureFetcher) Fetch() []bpfMapPressure {
+	results := []bpfMapPressure{}
+	var id ebpf.MapID = 0
+	var err error
+	for {
+		id, err = ebpf.MapGetNextID(id)
+		if errors.Is(err, os.ErrNotExist) {
+			break
+		}
+		if err != nil {
+			_ = logger.Warn("failed to get next map id", map[string]interface{}{
+				"id":        id,
+				log.FnError: err,
+			})
+			return nil
+		}
+		m, err := ebpf.NewMapFromID(id)
+		if err != nil {
+			_ = logger.Warn("failed to get map", map[string]interface{}{
+				"id":        id,
+				log.FnError: err,
+			})
+			return nil
+		}
+		info, err := m.Info()
+		if err != nil {
+			_ = logger.Warn("failed to get map info", map[string]interface{}{
+				"id":        id,
+				log.FnError: err,
+			})
+			return nil
+		}
+		for _, str := range f.mapNameStrings {
+			if strings.Contains(info.Name, str) {
+				itr := m.Iterate()
+				var key, value []byte
+				cnt := 0
+				for itr.Next(&key, &value) {
+					cnt++
+				}
+				results = append(results, bpfMapPressure{
+					mapId:       uint32(id),
+					mapName:     info.Name,
+					mapPressure: float64(cnt) / float64(info.MaxEntries),
+				})
+				break
+			}
```

Comment:
I think this block can be done concurrently (but it may cause a problem if the number of processes is too large, so we should control it with WaitGroup).

Reply:
The process of dumping and counting doesn't need to be concurrent because one execution per map is enough.
It seems inefficient that we use `strings.Contains()` every time in the for loop, but I think it's OK in this case because it's a temporary implementation and the number of `mapNames` is not quite large.

Reply:
@tkna 

> It seems inefficient that we use strings.Contains() every time in the for loop,

I don't think so. Is there any way to avoid using `strings.Contains()`?

> I think it's OK in this case because it's a temporary implementation and the number of mapNames is not quite large.

I think we should process them in parallel if there is no problem.
However, yes, as you say, we can skip it until we have to do it.